extensions/shared/external-media/sources/request-external-access/index.js

```diff
@@ -0,0 +1,31 @@
+/** @format */
+
+/**
+ * Internal dependencies
+ */
+
+import PopupMonitor from './popup-monitor';
+
+// Note: this is taken from Calypso
+const requestExternalAccess = ( url, cb ) => {
+	const popupMonitor = new PopupMonitor();
+	let lastMessage;
+
+	popupMonitor.open(
+		url,
+		null,
+		'toolbar=0,location=0,status=0,menubar=0,' + popupMonitor.getScreenCenterSpecs( 780, 700 )
+	);
+
+	popupMonitor.once( 'close', () => {
+		let keyringId = null;
+		if ( lastMessage && lastMessage.keyring_id ) {
+			keyringId = Number( lastMessage.keyring_id );
+		}
+		cb( keyringId );
+	} );
+
+	popupMonitor.on( 'message', message => ( lastMessage = message ) );
+};
+
+export default requestExternalAccess;
```

Comment:
Is this something we could pull from a package in the Calypso monorepo instead? It seems this could be made a package there maybe?
https://github.com/Automattic/wp-calypso/blob/ba88abde32eb957ad82ecab437f0fec491f45754/client/lib/sharing/index.js

Reply:
This sounds like a good way, exploring…

Reply:
Created issue: https://github.com/Automattic/wp-calypso/issues/41987